### PR TITLE
Add config to STAR and expand on GATK config

### PR DESCRIPTION
**Pull request by ihodes**
This adds configuration to STAR from the pipeline on down.

This also adds typed malformed read filters into the GATK configuration.

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/biokepi/237)

### CODE REVIEW

File: src/bfx_tools/gatk.ml [hunk 1]
```diff
@@ -6,42 +6,82 @@ module Remove = Workflow_utilities.Remove
 
 module Configuration = struct
 
-  module Indel_realigner = struct
-    include Tool_parameters
+  module Gatk_config = struct
+    type t = {
+      (* The name of the configuration, specific to Ketrew. *)
```
Comment (smondet) [inline src/bfx_tools/gatk.ml]: it's specific to Biokepi, not Ketrew :)

File: src/bfx_tools/gatk.ml [hunk 2]
```diff
@@ -6,42 +6,82 @@ module Remove = Workflow_utilities.Remove
...
+      name: string;
 
-    let default = {
-      name = "default";
-      parameters = [];
+      (* MalformedReadFilter options.
```
Comment (smondet) [inline src/bfx_tools/gatk.ml]: Also you can use ocamldoc comments there (`(** .. *)`).

File: src/bfx_tools/gatk.ml (hunk 1 above)
Comment (ihodes) [inline src/bfx_tools/gatk.ml]: Good catch, fixed :)

File: src/bfx_tools/gatk.ml (hunk 2 above)
Comment (ihodes) [inline src/bfx_tools/gatk.ml]: Gtk, fixed

**Comment by ihodes** [comment]
This is working & ready to merge.

**Pull request merged**